File: src/main/java/com/lumiyaviewer/lumiya/render/glres/buffers/GLVertexArrayObject.java
package com.lumiyaviewer.lumiya.render.glres.buffers;

import android.annotation.TargetApi;
import android.opengl.GLES30;
import com.lumiyaviewer.lumiya.render.glres.GLGenericResource;
import com.lumiyaviewer.lumiya.render.glres.GLResourceManager;
import com.lumiyaviewer.lumiya.render.glres.GLResourceManager.GLGenericResourceReference;

@TargetApi(18)
public class GLVertexArrayObject implements GLGenericResource {
    public final int size;
    private final int[] vaoIndices;

    private static class GLVertexArrayObjectReference extends GLGenericResourceReference {
        private final int[] vaoIndices;

        GLVertexArrayObjectReference(GLGenericResource gLGenericResource, GLResourceManager gLResourceManager, int[] iArr) {
            super(gLGenericResource, gLResourceManager);
            this.vaoIndices = iArr;
        }

        public void GLFree() {
            GLES30.glDeleteVertexArrays(this.vaoIndices.length, this.vaoIndices, 0);
        }
    }

    @TargetApi(18)
    public GLVertexArrayObject(GLResourceManager gLResourceManager, int i) {
        this.size = i;
        this.vaoIndices = new int[i];
        GLES30.glGenVertexArrays(i, this.vaoIndices, 0);
        GLVertexArrayObjectReference gLVertexArrayObjectReference = new GLVertexArrayObjectReference(this, gLResourceManager, this.vaoIndices);
    }

    public void Bind(int i) {
        GLES30.glBindVertexArray(this.vaoIndices[i]);
    }

    public void Unbind() {
        GLES30.glBindVertexArray(0);
    }
}

File: Linkpoint/src/main/app_broken_backup/com/linkpoint/render/glres/buffers/GLVertexArrayObject.kt
package com.linkpoint.render.glres.buffers

import android.annotation.TargetApi
import android.opengl.GLES30
import com.linkpoint.render.glres.GLGenericResource
import com.linkpoint.render.glres.GLResourceManager

@TargetApi(18)
class GLVertexArrayObject : GLGenericResource {
    Int size
    private IntArray vaoIndices

    private class GLVertexArrayObjectReference : GLResourceManager.GLGenericResourceReference {
        private IntArray vaoIndices

        GLVertexArrayObjectReference(GLGenericResource gLGenericResource, GLResourceManager gLResourceManager, IntArray iArr) {
            super(gLGenericResource, gLResourceManager)
            this.vaoIndices = iArr
        }

        fun GLFree()  {
            GLES30.glDeleteVertexArrays(this.vaoIndices.length, this.vaoIndices, 0)
        }
    }

    @TargetApi(18)
    constructor(gLResourceManager: GLResourceManager, i: Int) {
        this.size = i
        this.vaoIndices = Int[i]
        GLES30.glGenVertexArrays(i, this.vaoIndices, 0)
        GLVertexArrayObjectReference(this, gLResourceManager, this.vaoIndices)
    }

    fun Bind(i: Int)  {
        GLES30.glBindVertexArray(this.vaoIndices[i])
    }

    fun Unbind()  {
        GLES30.glBindVertexArray(0)
    }
}
